src/main/java/uk/gov/companieshouse/acsp/controllers/PaymentsController.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.companieshouse.acsp.controllers;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.acsp.Exception.ServiceException;
+import uk.gov.companieshouse.acsp.service.PaymentsService;
+import uk.gov.companieshouse.acsp.service.TransactionService;
+import uk.gov.companieshouse.api.model.payment.PaymentApi;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.sdk.manager.ApiSdkManager;
+
+@RestController
+public class PaymentsController {
+    @Autowired
+    private TransactionService transactionService;
+
+    @Autowired
+    private PaymentsService paymentsService;
+
+    @PostMapping(value = "/transaction/payments/{id}")
```

Comment:
need to change the url. Id being passed id is transactionId so "/transaction/{id}/payments" makes more sense, as currently it looks like you are passing paymentId

Reply:
comments addressed